scipy/linalg/_decomp_cossin.py

```diff
@@ -0,0 +1,194 @@
+from collections.abc import Iterable
+
+import numpy as np
+from numpy.linalg import LinAlgError
+
+from scipy._lib._util import _asarray_validated
+from scipy.linalg import block_diag
+from .lapack import _compute_lwork, get_lapack_funcs
+
+__all__ = ['cossin']
+
+
+def cossin(X, partitioning=None, compute_u=True, compute_vh=True):
+    """
+    Compute the cosine-sine (CS) decomposition of an `(m, m)` matrix.
+
+    The matrix X is a unitary or an orthogonal matrix, partitioned to X11, X12,
+    X21, X22 subblocks and the decomposition satisfies the following equality:
+
+    .. code-block:: none
```

Comment:
You can remove this rubric here and double the colons at end of the line above `equality::` to mark the following 4 space indented block as ascii code block.

Reply:
That's cool, thank you!